### DISCUSSION TITLE
Attaching nextcloud-aio-apache to extra network

### DISCUSSION THREAD

**Discussion by Javex** [nextcloud/all-in-one#4714] [Questions] [answered]
Hi! I've been running AIO for almost a year now and it's been going really well. Recently, I decided to switch my reverse proxy from nginx to Caddy and at the same time make my reverse proxy run inside Docker and no longer on the host system.

To achieve this, I have created a separate `reverse-proxy` network that all services get attached to with a section like this in their respective `docker-compose.yml`:

```
services:
  example:
    ...
    networks:
      - reverse-proxy
networks:
  reverse-proxy:
    external: true
```

Obviously, this doesn't work for AIO, because it creates and manages containers (and the network) on my behalf. I had a look at [the code](https://github.com/nextcloud/all-in-one/blob/main/php/src/Docker/DockerActionManager.php#L459), and I don't think I can change this, but I might be looking in the wrong place. Is it possible to achieve this?

My current workaround is to instead add my `caddy` container to the `nextcloud-aio`. I could also add a second "host" network for Caddy so that it can reach the port on the host like before.

Before I go down this road, are those the best options or am I overlooking something?

I know I could go down the manual road, but I love how easy AIO makes it to run Nextcloud!

**Comment by szaimen** [comment, marked as answer]
> My current workaround is to instead add my `caddy` container to the `nextcloud-aio`. I could also add a second "host" network for Caddy so that it can reach the port on the host like before.

Sounds like the solutions I would recommend to you 👍
